shared/chat/conversation/list-area/normal/container.js

```diff
@@ -32,29 +37,52 @@ const mapStateToProps = (state, {conversationIDKey}: OwnProps) => {
 }
 
 const mapDispatchToProps = (dispatch, {conversationIDKey}: OwnProps) => ({
-  _loadMoreMessages: () => dispatch(Chat2Gen.createLoadOlderMessagesDueToScroll({conversationIDKey})),
+  _loadNewerMessages: () => dispatch(Chat2Gen.createLoadNewerMessagesDueToScroll({conversationIDKey})),
+  _loadOlderMessages: () => dispatch(Chat2Gen.createLoadOlderMessagesDueToScroll({conversationIDKey})),
   _markInitiallyLoadedThreadAsRead: () =>
     dispatch(Chat2Gen.createMarkInitiallyLoadedThreadAsRead({conversationIDKey})),
   copyToClipboard: text => dispatch(ConfigGen.createCopyToClipboard({text})),
+  onJumpToRecent: () => dispatch(Chat2Gen.createJumpToRecent({conversationIDKey})),
 })
 
 const mergeProps = (stateProps, dispatchProps, ownProps: OwnProps) => ({
-  _loadMoreMessages: dispatchProps._loadMoreMessages,
+  _loadNewerMessages: dispatchProps._loadNewerMessages,
+  _loadOlderMessages: dispatchProps._loadOlderMessages,
+  centeredOrdinal: stateProps.centeredOrdinal,
+  containsLatestMessage: stateProps.containsLatestMessage,
   conversationIDKey: stateProps.conversationIDKey,
   copyToClipboard: dispatchProps.copyToClipboard,
   editingOrdinal: stateProps.editingOrdinal,
   lastMessageIsOurs: stateProps.lastMessageIsOurs,
   markInitiallyLoadedThreadAsRead: dispatchProps._markInitiallyLoadedThreadAsRead,
   messageOrdinals: stateProps.messageOrdinals.toList(),
   onFocusInput: ownProps.onFocusInput,
+  onJumpToRecent: dispatchProps.onJumpToRecent,
   scrollListDownCounter: ownProps.scrollListDownCounter,
+  scrollListToBottomCounter: ownProps.scrollListToBottomCounter,
   scrollListUpCounter: ownProps.scrollListUpCounter,
 })
 
 // We load the first thread automatically so in order to mark it read
 // we send an action on the first mount once
 let markedInitiallyLoaded = false
 
+const loadMoreMessages = (state, props, loadFn) => ordinal => {
+  if (state._conversationIDKey === props.conversationIDKey) {
+    if (state._lastLoadMoreOrdinalTime + 1000 > Date.now()) {
+      // ignore a load if its too recent for the same ordinal
```

Comment:
so should there be `&& ordinal === state.lastLoadMoreOrdinal`? also can you combine these `if`s?

Reply:
I just moved this around, I didn't really change anything here.